### PR TITLE
upgrade React to v18

### PR DESCRIPTION
**Pull request by aidan-starke**
## Description

 - upgrade React to v18.1.0
 - upgrade `@cennznet/api` to v2.1.1-alpha.2 and use `cvmToAddress` function
 - upgrade other dependencies
 - update all components & providers

### CODE REVIEW

File: libs/providers/FaucetProvider.tsx [hunk 1]
```diff
@@ -23,11 +20,9 @@ interface FaucetContextType {
 
 const FaucetContext = createContext<FaucetContextType>({} as FaucetContextType);
 
-interface FaucetProviderProps {}
+interface FaucetProviderProps extends PropsWithChildren {}
 
-function FaucetProvider({
-	children,
-}: PropsWithChildren<FaucetProviderProps>): ReactElement<FaucetProviderProps> {
+const FaucetProvider = ({ children }: FaucetProviderProps) => {
```
Comment (ken-futureverse) [inline libs/providers/FaucetProvider.tsx]: redundant, but can do the same `FC` here

**Review by ken-futureverse — APPROVED** [review]
just a few nitpicks but happy for it to merge, have you had a good test and there were no double rendering?

**Comment by aidan-starke** [comment]
> just a few nitpicks but happy for it to merge, have you had a good test and there were no double rendering?

found a double render of `AccountIdenticon` if `fadeOnChange` was true, we weren't using it, but I updated it to true and used `useDeferredValue` to avoid the double render

https://17.reactjs.org/docs/concurrent-mode-reference.html#usedeferredvalue

**Comment by ken-futureverse** [comment]
[React 18 Upgrade for Faucet](https://trello.com/c/qIL0en8d/46-react-18-upgrade-for-faucet)

**Review by ken-futureverse — APPROVED** [review]
The Identicon has always render double on Fast Reload, so it might not be React 18 related. Interesting that `useDefferedValue` fixed it. Nice one!

Can you port this change to App Hub also please 🙏

**Pull request merged**